Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -402,6 +403,57 @@ void GraphicsContextSkia::drawSkiaPath(const SkPath& path, SkPaint& paint)
         endTransparencyLayer();
 }
 
+bool GraphicsContextSkia::drawPathAsSingleElement(const Path& path, SkPaint& paint)
+{
+    if (hasDropShadow())
+        return false;
+
+    auto segment = path.singleSegment();
+    if (!segment)
+        return false;
+
+    auto drawArc = [&](const FloatPoint& center, float radius, float startAngle, float endAngle, bool isClosedArc) -> bool {
+        if (radius < 1)
+            return false;
+
+        auto sweepAngle = endAngle - startAngle;
+        SkScalar startDegrees = SkFloatToScalar(startAngle * 180 / std::numbers::pi_v<float>);
+        SkScalar sweepDegrees = SkFloatToScalar(sweepAngle * 180 / std::numbers::pi_v<float>);
+        static constexpr SkScalar s360 = SkIntToScalar(360);
+        bool isClosedOval = SkScalarNearlyEqual(std::abs(sweepDegrees), s360);
+        if (isClosedArc && !isClosedOval)
+            return false;
+
+        SkRect oval = { center.x() - radius, center.y() - radius, center.x() + radius, center.y() + radius };
+        if (isClosedArc)
+            m_canvas.drawOval(oval, paint);
+        else
+            m_canvas.drawArc(oval, startDegrees, sweepDegrees, false, paint);
+
+        return true;
+    };
+
+    bool handled = false;
+    WTF::switchOn(segment->data(),
+        [&](const PathArc& arc) {
+            handled = drawArc(arc.center, arc.radius, arc.startAngle, arc.endAngle, false);
+        },
+        [&](const PathClosedArc& closedArc) {
+            handled = drawArc(closedArc.arc.center, closedArc.arc.radius, closedArc.arc.startAngle, closedArc.arc.endAngle, true);
+        },
+        [&](const PathDataLine& line) {
+            if (paint.getStyle() != SkPaint::kFill_Style)
```

Comment:
is this intentional? `handled` will be true even if `kFill_Style` is used and a line is not drawn